### ISSUE TITLE
Nunchaku global hook can slow down workflows that don't even use the flowmatch scheduler and or cause other problems.

### ISSUE THREAD

**Issue by EricRollei** [erosDiffusion/ComfyUI-EulerDiscreteScheduler#9]
The nunchaku_compat.py in ComfyUI-EulerDiscreteScheduler is a compatibility layer for the MIT Han Lab Nunchaku quantization library.

Why is it causing problems?
Looking at the changelog:

1.0.7: nunchaku qwen patch fix, tiled diffusion patch fix users reported issues with dimensions not being handled correctly

The nunchaku_compat.py file patches [torch.nn.Module.__call__](vscode-file://vscode-app/c:/Users/Eric/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/workbench.html) globally to provide compatibility with Nunchaku-quantized models. This means every single PyTorch module forward pass in your entire ComfyUI session goes through their wrapper.

The Problem
This global hook is likely:

Interfering with GroundingDINO/SAM2's BERT embeddings
Possibly conflicting with our Hunyuan patches
Causing the 'weight' must be 2-D error

**Comment by erosDiffusion** [comment]
@EricRollei I will look into it, thanks for reporting. 

Do you have a flow that breaks the aforementioned features that I can start the investigation with ?
so far on my daily uses i have not encountered these errors (altough I don't use dino nor hunyan)

**Comment by EricRollei** [comment]
If you don't have Hunyuan Image 3 then my workflow might not work for you but you can examine it.   

[Unified-HI3-test_2025-12-15_17-04-49_000.json](https://github.com/user-attachments/files/24223920/Unified-HI3-test_2025-12-15_17-04-49_000.json)

I tried it with Marigold Depth, Depth Anything, Depth Anything v2, and one other depth node I have.  It's throwing the error on all of them.  Global hooks also seem to depend on load order and as far as I can tell the load order for custom nodes is alphabetical .
